cmd/namespace/validate_test.go

```diff
@@ -0,0 +1,115 @@
+package namespace
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/ory/x/cmdx"
+	"github.com/ory/x/configx"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+const fileMode = 0644
+
+const (
+	configEmbeddedYaml = `
+dsn: memory
+namespaces:
+  - name: testns0
+    id: 0
+  - name: testns1
+    id: 1
+`
+
+	configEmbeddedJson = `{"dsn":"memory","namespaces":[{"name":"testns0","id":0}]}`
+
+	configEmbeddedToml = `
+[[namespaces]]
+name = "testns0"
+id = 0
+
+[[namespaces]]
+name = "testns1"
+id = 1
+`
+)
+
+const configReference = `
+dsn: memory
+namespaces: %s
+`
+
+const nsyaml = "name: testns0\nid: 0"
+const nsjson = `{"name": "testns1", "id": 1}`
+
+func TestValidateConfigNamespaces(t *testing.T) {
+	cmd := cmdx.CommandExecuter{New: validateCommand}
+
+	t.Run("case=read valid config with embedded namespaces", func(t *testing.T) {
+		dir := t.TempDir()
+		fnyaml := dir + "/keto.yaml"
+		assert.NoError(t, ioutil.WriteFile(fnyaml, []byte(configEmbeddedYaml), fileMode))
+
+		fnjson := dir + "/keto.json"
+		assert.NoError(t, ioutil.WriteFile(fnjson, []byte(configEmbeddedJson), fileMode))
+
+		fntoml := dir + "/keto.toml"
+		assert.NoError(t, ioutil.WriteFile(fntoml, []byte(configEmbeddedToml), fileMode))
+
+		stdOut := cmd.ExecNoErr(t, "validate", "-c", fnyaml+","+fnjson+","+fntoml)
+		assert.Equal(t, "Congrats, all files are valid!\n", stdOut)
+	})
+
+	t.Run("case=config passed as varg fails", func(t *testing.T) {
+		fn := t.TempDir() + "/keto.yaml"
+		assert.NoError(t, ioutil.WriteFile(fn, []byte(configEmbeddedYaml), fileMode))
+
+		cmd.ExecExpectedErr(t, "validate", fn)
+	})
+
+	t.Run("case=read config with invalid embedded namespace", func(t *testing.T) {
+		fn := t.TempDir() + "/keto.yaml"
+		assert.NoError(t, ioutil.WriteFile(fn, []byte(`{"namespaces":[{"id":"x","name":"x"}]}`), fileMode))
+
+		stdOut := cmd.ExecExpectedErr(t, "validate", "--config", fn)
+		assert.Contains(t, stdOut, "not a valid namespace file")
```

Comment:
Also here, it would be nice to at least check if the invalid key is included in the output.
```suggestion
		assert.Contains(t, stdOut, "not a valid namespace file")
		assert.Contains(t, stdOut, "id")
```